File: app/src/main/java/com/taitsmith/busboy/api/WaypointResponse.java
package com.taitsmith.busboy.api;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class WaypointResponse {
    @SerializedName("Patterns")
    @Expose
    private final List<Pattern> patterns = null;

    public List<Pattern> getPatterns() {
        return patterns;
    }

    public static class Pattern {

        @SerializedName("Waypoints")
        @Expose
        private final List<Waypoint> waypoints = null;

        public List<Waypoint> getWaypoints() {
            return waypoints;
        }

        public static class Waypoint {
            @SerializedName("Latitude")
            @Expose
            private Double latitude;
            @SerializedName("Longitude")
            @Expose
            private Double longitude;

            public LatLng getLatLng() {
                return new LatLng(latitude,longitude);
            }
        }
    }
}

File: app/src/main/java/com/taitsmith/busboy/api/WaypointResponse.kt
package com.taitsmith.busboy.api

import com.google.android.gms.maps.model.LatLng
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class WaypointResponse {
    @SerializedName("Patterns")
    @Expose
    val patterns: List<Pattern>? = null

    class Pattern {
        @SerializedName("Waypoints")
        @Expose
        val waypoints: List<Waypoint>? = null

        class Waypoint {
            @SerializedName("Latitude")
            @Expose
            private val latitude: Double? = null

            @SerializedName("Longitude")
            @Expose
            private val longitude: Double? = null

            val latLng: LatLng
                get() = LatLng(latitude!!, longitude!!)
        }
    }
}